pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000.go

```diff
@@ -89,7 +121,24 @@ func (r *Rule2000) Run(ctx context.Context) (rule.RuleResult, error) {
 	case authenticationConfig.Anonymous == nil:
 		return rule.Result(r, rule.PassedCheckResult("Anonymous authentication is not enabled for the kube-apiserver.", configMapTarget)), nil
 	case authenticationConfig.Anonymous.Enabled:
-		return rule.Result(r, rule.FailedCheckResult("Anonymous authentication is enabled for the kube-apiserver.", configMapTarget)), nil
+		if r.Options == nil || len(authenticationConfig.Anonymous.Conditions) == 0 {
+			return rule.Result(r, rule.FailedCheckResult("Anonymous authentication is enabled for the kube-apiserver.", configMapTarget)), nil
+		}
+
+		var checkResults []rule.CheckResult
+
+		for _, condition := range authenticationConfig.Anonymous.Conditions {
+			if !slices.ContainsFunc(r.Options.AllowedEndpoints, func(allowedPath AllowedEndpoint) bool {
+				return allowedPath.Path == condition.Path
+			}) {
+				checkResults = append(checkResults, rule.FailedCheckResult(fmt.Sprintf("Anonymous authentication is not allowed for endpoint %s of the kube-apiserver.", condition.Path), configMapTarget))
```

Comment:
```suggestion
                                 endpointTarget := configMapTarget.With("details", fmt.Sprintf("endpoint: %s", condition.Path))
				checkResults = append(checkResults, rule.FailedCheckResult("Anonymous authentication is enabled for specific endpoints of the kube-apiserver.", endpointTarget))
```

I think it would be better if we move the specific endpoint to the target in `details`, since we use the message as a key for grouping the findings in the HTML report.

We should also do this for the accepted check.